### PR TITLE
fix: Filter names overflow wrap

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
Fixes a problem with the overflow wrap of filter names.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
https://github.com/apache/superset/assets/70410625/927989cd-a3a6-4e95-80d5-3b68a14e9fa2

https://github.com/apache/superset/assets/70410625/8055d8a7-e609-4b0e-b2c0-fd72c6f64d04

### TESTING INSTRUCTIONS
Check the videos for instructions.

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by kgabryje** [comment]
> The problem is that the FilterCard is already a tooltip and if we truncate the metric name we can't identify the metric.

I think this was the behaviour before the regression with overflowing title:

image

(added `max-width: 100%` to parent component to trigger the overflow). Could you elaborate why that won't work?

**Comment by michael-s-molina** [comment]
I thought we were avoiding tooltip on top of tooltip but it seems that's not the case given that Depend On will also show a tooltip. I'll add `max-width: 100%` to keep the previous behavior.

**Comment by kgabryje** [comment]
> I thought we were avoiding tooltip on top of tooltip but it seems that's not the case given that Depend On will also show a tooltip. I'll add `max-width: 100%` to keep the previous behavior.

I think we treat FilterCard more like a popover and given it has very different styling than a tooltip it's ok for us to use both here

**Comment by michael-s-molina** [comment]
@kgabryje I roll back `FilterCard` to its previous behavior with the addition of `overflow:hidden`. `max-width` does not respect `FilterCard`'s padding. Now it looks like:

Screenshot 2023-08-28 at 10 14 37

File: superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/FilterControl.tsx [hunk 1]
```diff
@@ -43,7 +43,7 @@ const VerticalFilterControlTitle = styled.h4`
   font-size: ${({ theme }) => theme.typography.sizes.s}px;
   color: ${({ theme }) => theme.colors.grayscale.dark1};
   margin: 0;
-  overflow-wrap: break-word;
+  overflow-wrap: anywhere;
```
[the remark this answers is not in the document]
Comment (michael-s-molina) [reply superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/FilterControl.tsx:46]: Yes, this correctly breaks the filter names in the native filters panel. The names are not truncated in that panel.

**Comment by kgabryje** [comment]
> @kgabryje I roll back `FilterCard` to its previous behavior with the addition of `overflow:hidden`. `max-width` does not respect `FilterCard`'s padding. Now it looks like:
> 
> Screenshot 2023-08-28 at 10 14 37

Great thank you! It seems unrelated to your fix, but the behaviour of displaying the tooltip seems flaky - when I open a filter card for the first time, the truncated title shows tooltip on hover, but when I close the filter card and open it again the tooltip doesn't appear anymore. Let's keep that in mind to fix it soon

**Review by kgabryje — APPROVED** [review]

**Pull request merged**